File: Arrays/128-Longest-Consecutive-Sequence/JavaScript/128-Longest-Consecutive-Sequence-Brute Force.js
/**
 * Problem: Longest Consecutive Sequence
 * Approach: Brute Force
 *
 * For every number, repeatedly check whether the next consecutive
 * number exists using linear search.
 *
 * Time Complexity: O(n^2)
 * Space Complexity: O(1)
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let longest = 0;

    for (const num of nums) {
        let x = num;
        let count = 1;

        while (nums.includes(x + 1)) {
            x++;
            count++;
        }

        longest = Math.max(longest, count);
    }

    return longest;
};

File: Arrays/128-Longest-Consecutive-Sequence/Python/128-Longest-Consecutive-Sequence-Brute Force.py
"""
Problem: Longest Consecutive Sequence
Approach: Brute Force

For every number, repeatedly check whether the next consecutive
number exists using linear search.

Time Complexity: O(n^2)
Space Complexity: O(1)
"""

class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        longest = 0

        for num in nums:
            x = num
            count = 1

            while x + 1 in nums:
                x += 1
                count += 1

            longest = max(longest, count)

        return longest